dts/bindings/pinctrl/nuvoton,npcx-pinctrl.yaml

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2022 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The Nuvoton pin controller is a singleton node responsible for controlling
+    pin function selection and pin properties. For example, you can use this
+    node to select peripheral pin functions.
+
+    Here is a list of supported standard pin properties:
+    - bias-disable: Disable pull-up/down (default behavior, not required).
+    - bias-pull-down: Enable pull-down resistor.
+    - bias-pull-up: Enable pull-up resistor.
+    - drive-push-pull: Output driver is push-pull (default, not required).
+    - drive-open-drain: Output driver is open-drain.
+    - output-high: Set output state high when pin configured.
+    - output-low: Set output state low when pin configured.
+
+    Custom pin properties for npcx series are available also:
+    - pinmux-locked: Lock pinmux configuration for peripheral device
+    - pinmux-gpio: Inverse pinmux back to gpio
+    - low-voltage-input: Select low-voltage input (1p8v) for io and peripheral devices
+
+    Driver strength and slew rate hardware defaults vary by SoC and pin.
+
+    An example for NPCX7 family, include the chip level pinctrl DTSI file in the
+    board level DTS:
+
+      #include <nuvoton/npcx/npcx7/npcx7-pinctrl.dtsi>
+
+    We want to use the I2C0_0 port of the NPCX7M6FB controller and enable the
+    internal 3.3V pull-up if its i2c frequency won't exceed 400kHz.
+
+    To change a pin's pinctrl default properties, add a reference to the
+    pin in the board's DTS file and set the properties as below:
+
+      &i2c0_0 {
+        pinctrl-0 = <&periph_alt2_i2c0_0_sl>;
+        pinctrl-names = "default";
+      }
+
+      &periph_alt2_i2c0_0_sl {
+        bias-pull-up; /* Enable internal pull-up for i2c0_0 */
+      };
+
+compatible: "nuvoton,npcx-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node.yaml
+      child-binding:
+        property-allowlist:
+          - bias-disable
+          - bias-pull-down
+          - bias-pull-up
+          - drive-push-pull
+          - drive-open-drain
+          - output-high
+          - output-low
+          - low-voltage-input
+          - pinmux-locked
+          - pinmux-gpio
+
+child-binding:
+    description: |
+      NPCX pin controller pin configuration state nodes.
+    properties:
+        pinmux:
+            type: array
+            required: false
+            description: Configuration of pinmux selection
+
+        io-pad:
+            type: array
+            required: false
+            description: IO identifier for pad configuration
+
+        periph-pupd:
+            type: array
+            required: false
+            description: Pull-up/down resistors on the corresponding peripheral devices only
```

Comment:
why is this required? can't this be inferred from the presense of `bias-pull-up/down`?

Reply:
Please refer to the first message of this PR. The IO-pads' functionalities such as open-drain, pull-up/down are different from gpio and peripheral devices in npcx series. And pinctrl functionalities of peripheral devices are scatted in different modules such as scfg, glue, gpio, and themselves. A new property to indicate the related control reg/bit of peripheral devices is necessary if ```bias-pull-up/down``` is present.